### ISSUE TITLE
Explore: "Load more" and "error occurred while loading images"

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#5120]
In Explore (both Featured and Uploaded via mobile) used to show more pictures simply by scrolling.

In master, instead a "Load more" button appears after the first 10 pictures. Even worse, tapping that button shows "error occurred while loading images" and does not load pictures.

The ideal behavior (which was happening until a month or two ago) is to endless scrolling: automatically load more images each time the user nears the bottom.

**Comment by bosankus** [comment]
Hi @nicolas-raoul can I work on this ?

**Comment by bosankus** [comment]
@nicolas-raoul this issue is visible on production app. In the beta flavour, am getting 4 fixed images. And no 'Load more' button. Can you suggest how can I recreate ?

**Comment by nicolas-raoul** [comment]
@bosankus Maybe you are using the testDebug flavor?
If yes, you should have more luck with the prodDebug flavor.

**Comment by bosankus** [comment]
Hi @nicolas-raoul , I was able to reproduce the issue.

I was able to see that in abstract class `PagingDataSource<T>` under method `performWithTryCatch(function: () -> Unit)`, I was receiving `java.lang.IllegalStateException: closed`. 

Reason:
- As per [documentation](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response-body/), if we use `response.body().string()` method then connection will be closed and response wont be consumed. 
- Also adding try block to store `rsp.body()` and throw IOException is probably a redundant step, when we are already getting the response body `rsp.peekBody(ERRORS_PREFIX.length()).`

Old Code I am referring to : https://github.com/commons-app/apps-android-commons/blob/6dec69147fa78d7ced3c4546a403ed59ca7e043b/app/src/main/java/fr/free/nrw/commons/OkHttpConnectionFactory.java#L89

New Changed code: #5122
